Wallet should not broadcast invalid transactions

philipr-za opened the pull request:
## Description
This PR just adds a check to make sure that broadcast protocols are not started for invalid transactions.

## How Has This Been Tested?
Unit test provided

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

delta1 commented on base_layer/wallet/tests/transaction_service/service.rs:
```diff
@@ -4653,6 +4653,64 @@ fn only_start_one_tx_broadcast_protocol_at_a_time() {
+
+    let tx_submit_calls =
+        runtime.block_on(rpc_service_state.wait_pop_submit_transaction_calls(1, Duration::from_secs(5)));
+    assert!(tx_submit_calls.is_err(), "Should not be no calls made");
```
accidental double negative? Or am I misreading

philipr-za replied:
yep copypasta

philipr-za committed:
Wallet should not broadcast invalid transactions

This PR just adds a check to make sure that broadcast protocols are not started for invalid transactions.
